rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/InlineItemModificationCreator.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.nodesource;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.ow2.proactive_grid_cloud_portal.rm.client.PluginDescriptor;
+import org.ow2.proactive_grid_cloud_portal.rm.client.nodesource.window.NodeSourceWindow;
+
+import com.smartgwt.client.widgets.form.fields.FormItem;
+import com.smartgwt.client.widgets.form.fields.RadioGroupItem;
+import com.smartgwt.client.widgets.form.fields.TextAreaItem;
+import com.smartgwt.client.widgets.form.fields.UploadItem;
+
+
+public class InlineItemModificationCreator {
+
+    public static final String EDIT_OR_UPLOAD_FORM_ITEM_SUFFIX = ".modify";
+
+    public static final String EDIT_FORM_ITEM_SUFFIX = ".edit";
+
+    public static final String EDIT_RADIO_OPTION_NAME = "edit";
+
+    public static final String UPLOAD_RADIO_OPTION_NAME = "upload";
+
+    private final NodeSourceWindow nodeSourceWindow;
+
+    public InlineItemModificationCreator(NodeSourceWindow nodeSourceWindow) {
+        this.nodeSourceWindow = nodeSourceWindow;
+    }
+
+    public List<FormItem> getModificationChoiceItemsForNonTextualFields(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+        FormItem chooseCredentialsFormItem;
+        List<FormItem> formItemsReplacingNonTextualFormItems = new LinkedList<FormItem>();
+
+        if (plugin.getPluginName().equals(nodeSourceWindow.getFocusedInfrastructurePluginName()) ||
+            plugin.getPluginName().equals(nodeSourceWindow.getFocusedPolicyPluginName())) {
+
+            RadioGroupItem editOrUploadFormItem = createRadioItemToModifyPluginField(plugin, pluginField);
+            formItemsReplacingNonTextualFormItems.add(editOrUploadFormItem);
+
+            TextAreaItem previousValueItem = createTextItemPrefilledWithFileContent(plugin, pluginField);
+            formItemsReplacingNonTextualFormItems.add(previousValueItem);
+
+            chooseCredentialsFormItem = createUploadItemDisabled(plugin, pluginField);
+
+        } else {
+
+            chooseCredentialsFormItem = new UploadItem(plugin.getPluginName() + pluginField.getName(),
+                                                       pluginField.getName());
+        }
+
+        nodeSourceWindow.addCredentialsPickerIcon(pluginField,
+                                                  formItemsReplacingNonTextualFormItems,
+                                                  chooseCredentialsFormItem);
+
+        return formItemsReplacingNonTextualFormItems;
+    }
+
+    private FormItem createUploadItemDisabled(PluginDescriptor plugin, PluginDescriptor.Field pluginField) {
+
+        FormItem chooseCredentialsFormItem;
```

Comment:
why declaring it empty here and not directly after?

Reply:
Wrong refactoring. Thanks